### PR TITLE
Feature/kyv 655 namespace configurations

### PR DESCRIPTION
**Pull request by severikupari1**

### CODE REVIEW

File: merchantapi/src/main/java/fi/hel/verkkokauppa/configuration/service/NamespaceService.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        isDuplicateNamespace(dto);
+        entity.setCreatedAt(DateTimeUtil.getFormattedDateTime());
+        NamespaceModel saved = namespaceRepository.save(entity);
+        return namespaceMapper.toDto(saved);
+    }
+
+    private void isDuplicateNamespace(NamespaceDto dto) {
+        if (namespaceRepository.findByNamespace(dto.getNamespace()).isPresent()) {
+            throw new CommonApiException(
+                    HttpStatus.NOT_FOUND,
```
Comment (wpamb) [inline merchantapi/src/main/java/fi/hel/verkkokauppa/configuration/service/NamespaceService.java:46]: Wrong error. Should probably be 400

File: merchantapi/src/main/java/fi/hel/verkkokauppa/configuration/repository/NamespaceRepository.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public interface NamespaceRepository extends ElasticsearchRepository<NamespaceModel, String> {
+    Optional<NamespaceModel> findByNamespace(String namespace);
+
+    List<NamespaceModel> findAllByNamespace(String namespace);
```
Comment (wpamb) [inline merchantapi/src/main/java/fi/hel/verkkokauppa/configuration/repository/NamespaceRepository.java:15]: Is there a reason it's even possible to have multiple entities with same namespace?

File: merchantapi/src/main/java/fi/hel/verkkokauppa/configuration/service/NamespaceService.java (hunk 1 above)
Comment (severikupari1) [reply merchantapi/src/main/java/fi/hel/verkkokauppa/configuration/service/NamespaceService.java:46]: Good point, I will change this to 400

File: merchantapi/src/main/java/fi/hel/verkkokauppa/configuration/repository/NamespaceRepository.java (hunk 2 above)
Comment (severikupari1) [reply merchantapi/src/main/java/fi/hel/verkkokauppa/configuration/repository/NamespaceRepository.java:15]: There can only be one namespace entity but there can be many merchants under same namespace.

Relation is one to many between namespace and merchants. Using same namespace for merchants creates possibility to create endpoint to change values for merchants in bulk

**Review by wpamb — APPROVED** [review]

**Pull request merged**